### PR TITLE
[WIP] run diff_schema.js on startup (fix)

### LINKED ISSUE #3161: diff schema can’t be run in wk docker image: node not found

### PR DESCRIPTION
**Pull request by jstriebel**
This makes the `diff_schema.js` run on wk startup, using [trireme](https://github.com/apigee/trireme).

Some compatibility changes were necessary, as trireme is built on old JS and node versions.

### Steps to test:
- Check in CI (e.g. smoke-test), that `diff_schema.js` is successfully run on startup.

### Issues:
- fixes #3161

------
- [ ] ~~Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- [ ] ~~Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- [ ] ~~Updated [documentation](../blob/master/docs) if applicable~~
- [ ] ~~Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: tools/postgres/diff_schema.js [hunk 1]
```diff
@@ -6,15 +6,14 @@ const execSync = require("child_process").execSync;
...
-const tmp = require("tmp");
 const rimraf = require("rimraf");
 const replace = require("replace-in-file");
 
 const POSTGRES_URL =
   typeof process.env.POSTGRES_URL !== "undefined"
     ? process.env.POSTGRES_URL
     : "jdbc:postgresql://localhost/webknossos";
-const scriptdir = __dirname;
+const scriptdir = "tools/postgres";
```
Comment (jstriebel) [inline tools/postgres/diff_schema.js]: we should rather use `__dirname` with a fallback (if I recall it correctly, the scriptName contains the full path in trireme)

File: tools/postgres/diff_schema.js [hunk 2]
```diff
@@ -25,9 +24,12 @@ function dump(parameter) {
...
     return dumpToFolder(POSTGRES_URL);
   } else {
-    const { dbName, dbHost, postgresUrl } = initTmpDB();
-    console.log(`Creating DB ${dbName}`);
-    execSync(`psql -U postgres -h ${dbHost} -c "CREATE DATABASE ${dbName};"`, {
+    const tmpdb = initTmpDB();
+    const dbName = tmpdb[0];
+    const dbHost = tmpdb[1];
+    const postgresUrl = tmpdb[2];
+    console.log("Creating DB " + dbName);
```
Comment (jstriebel) [inline tools/postgres/diff_schema.js]: Backticks unfortunately cause an `invalid character` exception.

**Comment by jstriebel** [comment]
@normanrz: Do you think the changes in the script are worth it? I think that's currently the easiest option to fix it. I also mentioned alternatives in the issue #3161.

File: tools/postgres/diff_schema.js [hunk 3]
```diff
@@ -49,43 +51,43 @@ function dump(parameter) {
...
 function dumpToFolder(postgresUrl) {
-  const tmpDir = tmp.dirSync();
+  const tmpDir = execSync("mktemp -d").toString().trim();
   try {
-    execSync(`${scriptdir}/dump_schema.sh ${tmpDir.name}`, { env: { POSTGRES_URL: postgresUrl } });
+    execSync(scriptdir + "/dump_schema.sh " + tmpDir, { env: { POSTGRES_URL: postgresUrl } });
   } catch (err) {
-    console.log(`CLEANUP: remove ${tmpDir.name}`);
-    rimraf.sync(tmpDir.name);
+    console.log("CLEANUP AAA: remove " + tmpDir);
```
Comment (philippotto) [inline tools/postgres/diff_schema.js]: Why "AAA"?

Comment (jstriebel) [reply tools/postgres/diff_schema.js]: sorry, that's just from my debugging, I'll remove it :D

**Comment by jstriebel** [comment]
@philippotto Maybe you can help me with that: To collect and build the assets, I need all node modules for `diff_schema.js` copied or installed in a special location. I tried to implement a copying solution here: https://github.com/scalableminds/webknossos/pull/3284/files#diff-798d4e806f4dee2aa886624fb8b2c1ebR101
This copies only the first layer of dependencies. Do you know a way, how to copy or install just some dependendencies and their subdependencies, maybe via yarn? Or how to get the subdependencies of a module?

**Comment by jstriebel** [comment]
@fm3 Could you please have a look at the scala code? Thanks :-)

**Pull request merged**